app/packages/spaces/src/components/Workspaces/WorkspaceEditor.tsx

```diff
@@ -96,40 +100,74 @@ export default function WorkspaceEditor() {
       <DialogActions sx={{ justifyContent: "space-between", px: 3 }}>
         <Box>
           {edit && (
-            <Button
-              startIcon={<Delete color="error" />}
+            <MuiButton
+              startIcon={<Delete color={isDeleting ? "disabled" : "error"} />}
               color="error"
               onClick={() => {
-                deleteOperator.execute({ name }, { skipOutput: true });
-                reset();
-                handleClose();
+                setStatus("deleting");
+                executeOperator(
+                  DELETE_WORKSPACE_OPERATOR,
+                  { name },
+                  {
+                    callback: (result) => {
+                      if (!result.error) {
```

Comment:
Is it so that you can use it here?

Reply:
Yah, I refactored the workspace code little bit to use `executeOperator` with the new callback instead of `useOperatorExecutor`